src/views/product-tutorials-view/helpers/heading-helpers.ts

```diff
@@ -0,0 +1,101 @@
+import slugify from 'slugify'
+import { TableOfContentsHeading } from 'layouts/sidebar-sidecar/components/table-of-contents'
+import { ProductPageBlock } from 'lib/learn-client/types'
+import { ProductViewBlock } from '../components/product-view-content'
+
+/**
+ * Given pageData for product view,
+ * as well as a productName,
+ *
+ * Return an array of Sidecar-compatible
+ * TableOfContentsHeading objects, each representing
+ * a section that will be rendered on the page.
+ */
+function buildLayoutHeadings(
+  pageData: {
+    blocks: ProductViewBlock[]
+    showProductSitemap?: boolean
+  },
+  productName: string
+): TableOfContentsHeading[] {
+  const { blocks, showProductSitemap } = pageData
+
+  /**
+   * Build an <h1> overview heading
+   */
+  const overviewHeading = [getOverviewHeading(productName)]
+
+  /**
+   * Extract headings from each block
+   */
+  const blockHeadings = blocks.reduce(
+    (acc: TableOfContentsHeading[], block: ProductViewBlock) => {
+      if (block.type !== 'CardList') {
+        acc.push({
+          title: block.heading,
+          slug: block.headingSlug,
+          level: 2,
+        })
+      }
+      return acc
+    },
+    []
+  )
+
+  /**
+   * If there's a product sitemap, build add a heading item for it
+   */
+  const sitemapHeading = showProductSitemap ? [getSitemapHeading()] : []
+
+  /**
+   * Flatten the array of headings, and return them
+   */
+  return [...overviewHeading, ...blockHeadings, ...sitemapHeading]
+}
+
+/**
+ * Given a productName,
+ * return a TableOfContentsHeading object
+ * representing a top-level heading for a product tutorials view
+ */
+function getOverviewHeading(productName: string): TableOfContentsHeading {
```

Comment:
I've split out `getOverviewHeading` and `getSitemapHeading` to make it more clear that they're used both server-side (to generate the array of `headings` for the sidecar), as well as client-side (to make sure the headings being linked to get rendered).